pkg/kubelet/config/common.go

```diff
@@ -73,6 +73,21 @@ func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.Node
 	// The generated UID is the hash of the file.
 	pod.Annotations[kubetypes.ConfigHashAnnotationKey] = string(pod.UID)
 
+	if isFile {
+		// Applying the default Taint tolerations to static pods,
+		// so they are not evicted when there are node problems.
```

Comment:
> I am a little bit unsure about this -- does this mean that we'll need to expand the list whenever new taints are added?

That's a good point. Yes, that's what it means. There's a way to say "tolerate all taints" (just leave the "key" field empty). If we think static pods will want to tolerate all NoExecute taints, then we just need one toleration, with no key (and the rest of the fields as-is).